pna_control/pna_control.py

```diff
@@ -114,7 +136,40 @@ def powersweep(startpower: float, endpower: float, numsweeps: int, centerf: floa
     file.write('POINTS: '+str(points)+'\n')
     file.close()
 
-    #run each sweep while increasing averages for each power
+    #run each sweep
     for i in sweeps:
-        getdata(centerf, span, temp, averages, i, edelay, ifband, points, outputfile)
+        get_data(centerf, span, temp, averages, i, edelay, ifband, points, outputfile)
+        # Iterate number of averages as power decreases
         averages = averages * ((10**(stepsize/10))**0.5)
+
+def name_datafile(outputfile,power,temp):
+    filename = outputfile+'_'+str(power)+'dB'+'_'+str(temp)+'mK'
+    filename = filename.replace('.','p')
+
+    return filename
+    
+def timestamp_folder(dir,meastype):
+    result = time.localtime(time.time())
+    output = str(result.tm_year)
+    if len(str(result.tm_mon)) < 2:
+        output = output + '0' + str(result.tm_mon)
+    else:
+        output = output + str(result.tm_mon)
+    if len(str(result.tm_mday)):
+        output = output + '0' + str(result.tm_mday) + '_' + str(result.tm_hour) + '_' + str(result.tm_min) + '_' + str(result.tm_sec)
+    else:
+        output = output + str(result.tm_mday) + '_' + str(result.tm_hour) + '_' + str(result.tm_min) + '_' + str(result.tm_sec)
```

Comment:
One concern I have of using timestamps in this manner is that for high power sweeps which can be completed in only a few averages, the code could produce more than one folder per second in which case the timestamps would be duplicate. I've had this problem before with the fit code, and the notation I've used is that if there is already a directory with the timestamp given, it will simply put a _2 (or _x for however many it may need) at the end of the timestamp.